### PR TITLE
feat(stats): add package nanmeanvar

### LINKED ISSUE #5575: [RFC]: add `stats/incr/nanmeanvar`
## Description

This RFC proposes adding a new package to the `stats/incr/*` namespace: `@stdlib/stats/incr/nanmeanvar`.

This package is similar to [`@stdlib/stats/incr/meanvar`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/incr/meanvar), except that it performs special handling of `NaN` values.

### Key Points

When implementing this package, you'll need to do the following:

#### Create package files

To see examples of similar packages, see [`@stdlib/stats/incr/meanvar`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/incr/meanvar) (propagates `NaN` values) and [`@stdlib/stats/incr/nansum`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/incr/nansum) (wraps a `NaN` propagating accumulator and skips `NaN` values).

In particular notice how `nansum` is a thin wrapper around `sum`. In most cases, this is what we are looking for.

While the package proposed in this RFC will **not** match the implementations found in the above packages, those packages should provide a conceptual idea of what is desired. Do **not** simply copy-paste the code found in those packages without reasoning about expected behavior and API design.

#### Update source files

Once you've created package files, proceed to update all tests, documentation, and examples to reflect the API and expected behavior for the package. You should spend some time studying the tests in [`@stdlib/stats/incr/meanvar`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/incr/meanvar). We expect this package to have equivalent and fully comprehensive tests. The primary difference will be in the handling of `NaN` values.

To see an example of what is expected, study the tests in

-   [`@stdlib/stats/incr/sum`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/incr/sum) and [`@stdlib/stats/incr/nansum`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/incr/nansum)

Compare and contrast the tests in that pair of packages.

#### Run tests and other commands

For each of the following commands, please run them from the root stdlib repository directory (not the package folder!).

To run unit tests,

```bash
make test TESTS_FILTER=".*/stats/incr/nanmeanvar/.*"
```

To run examples,

```bash
make examples EXAMPLES_FILTER=".*/stats/incr/nanmeanvar/.*"
```

To run benchmarks,

```bash
make benchmark BENCHMARKS_FILTER=".*/stats/incr/nanmeanvar/.*"
```

### Create pull request

Provided all tests, examples, and benchmarks successfully execute and pass and that you've updated the package's documentation, you are now ready to open a pull request!

* * *

## Notes

- If you are interested in contributing a PR which addresses this RFC and still getting familiar with our project conventions, please do **not** submit LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `[RFC]:`.

### PR DESCRIPTION
**Pull request by hans2001**
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves #5575 .

## Description

> What is the purpose of this pull request?

This pull request introduces the **nanmeanvar** package—a thin wrapper around **meanvar** that enhances robustness. It adds type checking so that if a NaN or non-numeric value is provided, the update is skipped and the accumulator continues to return the previously computed mean directional accuracy. Test cases have been updated to verify that these edge cases are correctly handled.

This pull request:

test coverage:
Screenshot 2025-03-19 at 6 21 20 PM

example:
Screenshot 2025-03-19 at 6 22 00 PM

benchmark:
Screenshot 2025-03-19 at 6 23 33 PM

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5575 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/incr/nanmeanvar/README.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# incrnanmeanvar
+
+> Compute an [arithmetic mean][arithmetic-mean] and an [unbiased sample variance][sample-variance] incrementally, ignoring `NaN` value.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/incr/nanmeanvar/README.md:23]: Singular "value" should be plural "values" for consistency with similar NaN-ignoring packages like `nanmean`.

```suggestion
> Compute an [arithmetic mean][arithmetic-mean] and an [unbiased sample variance][sample-variance] incrementally, ignoring `NaN` values.
```

File: lib/node_modules/@stdlib/stats/incr/nanmeanvar/README.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+## Usage
+
+```javascript
+var incrnanmeanvar = require( '@stdlib/stats/incr/nanmeanvar' );
+```
+
+#### incrnanmeanvar( \[out] )
+
+Returns an accumulator `function` which incrementally computes an [arithmetic mean][arithmetic-mean] and [unbiased sample variance][sample-variance], ignoring `NaN` value.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/incr/nanmeanvar/README.md:67]: Singular "value" should be plural "values".

```suggestion
Returns an accumulator `function` which incrementally computes an [arithmetic mean][arithmetic-mean] and [unbiased sample variance][sample-variance], ignoring `NaN` values.
```

File: lib/node_modules/@stdlib/stats/incr/nanmeanvar/docs/repl.txt [hunk 3]
```diff
@@ -0,0 +1,39 @@
+
+{{alias}}( [out] )
+    Returns an accumulator function which incrementally computes an arithmetic
+    mean and unbiased sample variance, ignoring `NaN` value.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/incr/nanmeanvar/docs/repl.txt:4]: Singular "value" should be plural "values".

```suggestion
    mean and unbiased sample variance, ignoring `NaN` values.
```

File: lib/node_modules/@stdlib/stats/incr/nanmeanvar/docs/types/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+/// <reference types="@stdlib/types"/>
+
+import { ArrayLike } from '@stdlib/types/array';
+
+/**
+* If provided a value, the accumulator function returns updated results. If not provided a value, the accumulator function returns the current results.
+*
+* ## Notes
+*
+* -   If provided `NaN`, the arithmetic mean and unbiased sample variance values are equal to `NaN` for all future invocations.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/incr/nanmeanvar/docs/types/index.d.ts:30]: **Incorrect NaN behavior documentation.** This note incorrectly states that NaN propagates to all future results, but the actual behavior is that NaN values are skipped/ignored and the accumulator continues with the current state.

```suggestion
* -   If provided `NaN`, the input value is ignored and the accumulator returns the current results.
```

File: lib/node_modules/@stdlib/stats/incr/nanmeanvar/docs/types/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+*
+* @param x - input value
+* @returns output array or null
+*/
+type accumulator = ( x?: number ) => ArrayLike<number> | null;
+
+/**
+* Returns an accumulator function which incrementally computes an arithmetic mean and unbiased sample variance, ignoring `NaN` value.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/incr/nanmeanvar/docs/types/index.d.ts:38]: Singular "value" should be plural "values".

```suggestion
* Returns an accumulator function which incrementally computes an arithmetic mean and unbiased sample variance, ignoring `NaN` values.
```

File: lib/node_modules/@stdlib/stats/incr/nanmeanvar/lib/main.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+
+var isNumber = require( '@stdlib/assert/is-number' ).isPrimitive;
+var incrmeanvar = require( '@stdlib/stats/incr/meanvar' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+
+
+// MAIN //
+
+/**
+* Returns an accumulator function which incrementally computes an arithmetic mean and unbiased sample variance, ignoring `NaN` value.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/incr/nanmeanvar/lib/main.js:31]: Singular "value" should be plural "values".

```suggestion
* Returns an accumulator function which incrementally computes an arithmetic mean and unbiased sample variance, ignoring `NaN` values.
```

File: lib/node_modules/@stdlib/stats/incr/nanmeanvar/lib/index.js [hunk 7]
```diff
@@ -19,29 +19,35 @@
 'use strict';
 
 /**
-* Compute an arithmetic mean of absolute values incrementally, ignoring `NaN` values.
+* Compute an arithmetic mean and unbiased sample variance incrementally.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/incr/nanmeanvar/lib/index.js:22]: Module description should mention NaN handling to be consistent with the function description and other NaN-ignoring packages.

```suggestion
* Compute an arithmetic mean and unbiased sample variance incrementally, ignoring `NaN` values.
```

File: lib/node_modules/@stdlib/stats/incr/nanmeanvar/package.json [hunk 8]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "@stdlib/stats/incr/nanmeanabs",
+  "name": "@stdlib/stats/incr/nanmeanvar",
   "version": "0.0.0",
-  "description": "Compute an arithmetic mean of absolute values incrementally, ignoring NaN values.",
+  "description": "Compute an arithmetic mean and unbiased sample variance incrementally, ignoring `NaN` value.",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/incr/nanmeanvar/package.json:4]: Singular "value" should be plural "values".

**Review by Planeshifter — CHANGES_REQUESTED** [review]

**Review by Planeshifter — APPROVED** [review]
Let's land it.

Note: CI failure seems to be due to bug in doctest causing it to only comparing values after the entire example block has finished executing, which given return value is reference to same value causes only last one to be considered.
